src/markdown/developers/component-practices.md

```diff
@@ -0,0 +1,89 @@
+---
+path: "/guides/component-practices"
+title: "Component Practices"
+intro: "A guide to writing components."
+---
+
+## Component Development Practices
+
+This is a summary of practices to follow when writing components. We explain them in the context of using design-system components, but they can be applied wherever you use React.
+
+### Use Existing Properties first
+
+Look for props that can be used to implement specific behaviour. Remember that you can use a number of styled props to change the look of NDS components.
+For example, to update the styles of a Text instead of overriding with styled components:
+```javascript
+const FormLabel  = styled(Text)({
+        fontSize: "12px",
+        textTransform: "uppercase",
+        textAlign: "right"
+})
+```
+You can use the existing props to make the changes:
+
+```javascript
+const FormLabel = ({children}}) => (
+    <Text fontSize="small" textTransform="uppercase" textAlign="right">{children}:</Text>)
+``` 
+
+
+### Name components after what they do
+
+Name components after what they do, not how they do it. Often we're tempted to name an internal components Styled____ just for the purposed of making some style changes, but this makes it tricky to understand what that component was meant to do and tp madd other text components later on.
+
+Instead of StyledText, if the component has a specific purpose like displaying capacity values try a contextual name like CapacityText, or if its meant to be used as a Label for another value try CapacityLabel.
```

Comment:
What's the benefit of doing it this way vs the other? What future things get unlocked by naming it after its purpose, rather than how it looks?